app/controllers/api/v2/trends_controller.rb

```diff
@@ -12,20 +13,25 @@ class TrendsController < V2::BaseController
 
       api :GET, "/trends/", N_("List of trends counters")
       def index
+        Foreman::Deprecation.api_deprecation_warning("use /foreman_statistics/trends endpoind from Foreman Statistics plugin instead")
         @trends = resource_scope_for_index
+        render 'foreman_statistics/api/v2/trends/index'
```

Comment:
Should this controller maintain the action logic or should it only call the relevant methods in the plugin if it is installed?

Reply:
It would be much better if it'd call the relevant method, but I've not found an easy way to do so :thinking:

Reply:
Thinking more about it, since this API won't work anyways without the plugin, if we only want to maintain the old end points for compatibility in scripts - perhaps we should just drop it from core completely and make the old route point to the new controller in the plugin?

Reply:
Can core return a HTTP 501 Not Implemented unless the plugin is present? I think that's a good pattern to use for things we move out of core.

Reply:
Interesting, I've took both ideas, only issue is, now user don't get to see the deprecation warning :thinking:
Should we add the warning in the plugin and detect the path, that took user to that action?

Reply:
sounds good to me to add the deprecation warning in the plugin when using the old path

Reply:
Right, here: https://github.com/theforeman/foreman_statistics/pull/13